native.go

```diff
@@ -0,0 +1,130 @@
+package libovsdb
+
+import (
+	"fmt"
+)
+
+// NativeAPI is an API that offers functions to interact with libovsdb without
+// having to handle it's internal objects. It uses a DatabaseSchema to infer the
+// type of each value and make translations.
+// OvsMaps are translated to go maps with specificif key and values. I.e instead of
+//	having to deal with map[interface{}][interface{}], the user will be able to
+//	user  map[string] string (or whatever native type can hold the column value)
+// OvsSets, same as OvsMap.
+//
+// OvsUUID are translated to and from strings
+// If the column type is an enum, the native type associated with the underlying enum
+// type is used (e.g: string or int)
+// Also, type checkings are done. E.g: if you try to put an integer in a column that has
+// type string, the API will refuse to create the Ovs object for you
+type NativeAPI struct {
+	schema DatabaseSchema
+}
+
+// NewNativeAPI returns a NativeAPI
+func NewNativeAPI(schema DatabaseSchema) NativeAPI {
+	return NativeAPI{
+		schema: schema,
+	}
+}
+
+// GetRowData transforms a Row to a native type data map[string] interface{}
+func (na NativeAPI) GetRowData(tableName string, row *Row, result interface{}) error {
+	if row == nil {
+		return nil
+	}
+	return na.GetData(tableName, row.Fields, result)
+}
+
+// GetData transforms a map[string]interface{} containing OvS types (e.g: a ResultRow
+// has this format) to native.
+// The result object must be given as pointer to map[string] interface{}
+func (na NativeAPI) GetData(tableName string, ovsData map[string]interface{}, result interface{}) error {
+	table, ok := na.schema.Tables[tableName]
+	if !ok {
+		return fmt.Errorf("TableName not found in schema %s", tableName)
+	}
+	dataP, ok := result.(*map[string]interface{})
+	if !ok {
+		return fmt.Errorf("NativeAPI.GetData requires result to be *map[string]interface{}")
+	}
+	if *dataP == nil {
```

Comment:
looks strange, i think if cast is not success returned nil variable and ok == false
why we cast result to pointer to map ? why not map[string]interface{} ?

Reply:
I keep forgetting map is always passed by reference. I think we can pass map[string]interface{} and still modify the result value.

Reply:
Now I remember why I did this. If we don't accept a pointer to map, we cannot accept non-initialized maps, which is more convenient, e.g:
```
var mymap map[string] interface{}
GetData(&map)
```

Reply:
The whole idea of accepting a pointer as result was to be able to have the same API for NativeAPI (this one) and the future DynamicAPI.
However, now that I think about it, it makes little sense to have a common interface since both APIs will behave different. So I'll change the API to:
`
GetData(string, map[string]interface{}) (map[string]interface{}, error)
`